src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/npc_arthas.cpp

```diff
@@ -1095,28 +1095,17 @@ class npc_arthas_stratholme : public CreatureScript
                         me->GetCreatureListWithEntryInGrid(nearbyVictims, urand(0, 1) ? NPC_CITIZEN : NPC_RESIDENT, 60.0f);
                         if (!nearbyVictims.empty())
                         {
-                            std::list<Creature*>::iterator it = nearbyVictims.begin();
-                            std::advance(it, urand(0, nearbyVictims.size()-1));
-                            Emote emote;
-                            switch (urand(0, 3))
+                            std::vector<Emote> emotes
                             {
-                                case 0:
-                                    emote = EMOTE_ONESHOT_TALK;
-                                    break;
-                                case 1:
-                                    emote = EMOTE_ONESHOT_EXCLAMATION;
-                                    break;
-                                case 2:
-                                    emote = EMOTE_ONESHOT_RUDE;
-                                    break;
-                                case 3:
-                                    emote = EMOTE_ONESHOT_ROAR;
-                                    break;
-                                default:
-                                    break;
-                            }
-                            if ((*it)->IsAlive())
-                                (*it)->HandleEmoteCommand(emote);
+                                EMOTE_ONESHOT_TALK,
+                                EMOTE_ONESHOT_EXCLAMATION,
+                                EMOTE_ONESHOT_RUDE,
+                                EMOTE_ONESHOT_ROAR
+                            };
+
+                            Creature* victim = Trinity::Containers::SelectRandomContainerElement(nearbyVictims);
+                            if (victim->IsAlive())
+                                victim->HandleEmoteCommand(Trinity::Containers::SelectRandomContainerElement(emotes));
```

Comment:
`Trinity::Containers::SelectRandomContainerElement` works even on plain c arrays, no need to make this a vector

Reply:
Fixed. Thanks!